storage/boltz/store.go

```diff
@@ -22,7 +22,15 @@ import (
 	"strings"
 )
 
```

Comment:
Trying to make sure you can't create a child store with a different entity type than the parent store, as this will likely only lead to confusion and error